2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_queue/LinkedQueueTest.java

```diff
@@ -1,103 +1,382 @@
+            & (field.getName().contains("next"));
+
+    private static final Predicate<Field> ELEMENT_FIELD = field -> field
+            .getGenericType()
+            .getTypeName()
+            .equals("T")
+            & (field.getName().contains("elem")
+            | field.getName().contains("value")
+            | field.getName().contains("item"));
+
+    private static final Predicate<Field> NEXT_FIELD = field -> field
+            .getGenericType()
+            .getTypeName()
+            .equals("Node<T>")
+            & (field.getName().contains("next"));
+
+    private static final Predicate<Field> SIZE_FIELD = field -> field
+            .getType()
+            .getSimpleName()
+            .equals("int")
+            & (field.getName().equals(SIZE_NAME));
+
+    private static final Predicate<Field> HEAD_FIELD = field -> field
+            .getType().getSimpleName().equals(NODE_NAME)
+            & (field.getName().contains("head")
+            | field.getName().contains("first"));
+
+    private static final Predicate<Field> TAIL_FIELD = field -> field
+            .getType().getSimpleName().equals(NODE_NAME)
+            & (field.getName().contains("tail")
+            | field.getName().contains("last"));
 
     private Queue<Integer> integerQueue = new LinkedQueue<>();
 
+
     @Test
     @Order(1)
-    void sizeOfEmptyQueue() {
-        assertEquals(0, integerQueue.size());
+    void createNodeClass() {
+        Class<?> innerClass = getInnerStaticNodeClass();
+        String name = innerClass.getSimpleName();
+
+        assertThat(name).isEqualTo(NODE_NAME);
     }
 
     @Test
     @Order(2)
-    void size() {
-        integerQueue.add(98);
-        integerQueue.add(9);
-        integerQueue.add(5);
-        integerQueue.add(6);
+    void chekFieldsNameInNodeClass() {
+        Class<?> innerClass = getInnerStaticNodeClass();
+        boolean hasElementField = hasField(innerClass, ELEMENT_FIELD);
+        boolean hasNodeField = hasField(innerClass, NODE_FIELD);
 
-        assertEquals(4, integerQueue.size());
+        assertThat(hasElementField).isTrue();
+        assertThat(hasNodeField).isTrue();
     }
 
     @Test
     @Order(3)
-    void addElementIntoEmptyQueue() {
-        integerQueue.add(1312);
+    void checkFieldsInQueueCLass() {
+        Class<?> baseClass = this.integerQueue.getClass();
+        boolean hasHeadField = hasField(baseClass, HEAD_FIELD);
+        boolean hasSizeFiled = hasField(baseClass, SIZE_FIELD);
+        boolean hasTailFiled = hasField(baseClass, TAIL_FIELD);
 
-        assertEquals(1, integerQueue.size());
-        assertEquals(1312, integerQueue.poll().intValue());
+        assertThat(hasHeadField).isTrue();
+        assertThat(hasSizeFiled).isTrue();
+        assertThat(hasTailFiled).isTrue();
     }
 
     @Test
     @Order(4)
-    void addElement() {
-        integerQueue.add(324);
-        integerQueue.add(23);
-        integerQueue.add(5);
+    void checkSizeInEmptyQueue() {
```

Comment:
Lets adhere to tests name convention like this: `methodNameDoesSomethingWhenSomeCondition`.
For instance `sizeReturnsZeroWhenQueueIsEmpty`